workflows/diagnostics/fv3net/diagnostics/offline/compute_diagnostics.py

```diff
@@ -364,11 +365,22 @@ def variance_zonal_avg_3d(diag_arg, mask_type=mask_type):
         )
         if len(predicted) == 0:
             return xr.Dataset()
-        mean = weighted_mean(target, weights=grid.area, dims=HORIZONTAL_DIMS).mean(
-            "time"
-        )
+
+        zonal_mean = vcm.zonal_average_approximate(grid.lat, target).mean("time")
+
+        # Fill cubedsphere with zonal mean, interpolating between integer latitudes
+        zonal_mean_interp = xr.Dataset()
+        for var in zonal_mean:
+            da = zonal_mean[var]
+            # Ignore extrapolation errors for test case with randomized lat data
+            interp_func = interp1d(da.lat, da.transpose(..., "lat"), bounds_error=False)
+            zonal_mean_interp[var] = xr.DataArray(
+                interp_func(grid.lat.transpose(*HORIZONTAL_DIMS)),
+                dims=("pressure", *HORIZONTAL_DIMS),
+            )
```

Comment:
You should be able to do this without interpolating if you compute the variance as the mean of the squares minus the square of the means.

Reply:
Thanks good point. I'll just use the cloud_ml function above.